src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/RSSAnalyzer.scala

```diff
@@ -13,21 +13,40 @@ class RSSAnalyzer extends Analyzer[RSSEntry] with Serializable with AnalysisDefa
 
   override def toSchema(item: RSSEntry, locationFetcher: LocationFetcher, imageAnalyzer: ImageAnalyzer): ExtendedDetails[RSSEntry] = {
     val body = readDescription(item)
+    val url = getSourceUrlFromItem(item)
     ExtendedDetails(
-      eventid = s"RSS.${item.uri}",
+      eventid = s"RSS.${url.getOrElse(item.uri)}",
       sourceeventid = item.uri,
       eventtime = item.publishedDate,
       body = body,
       imageurl = None,
       title = item.title,
       externalsourceid = item.source.uri,
       pipelinekey = "RSS",
-      sourceurl = item.uri,
+      sourceurl = url.getOrElse(""),
       sharedLocations = List(),
       original = item
     )
   }
 
+  private[analyzer] def getSourceUrlFromItem(item: RSSEntry): Option[String] = {
```

Comment:
Should make these private not package private to avoid people from taking unexpected dependencies.